lib/portlayer/exec/container.go

```diff
@@ -109,15 +109,18 @@ type Container struct {
 	// Current state
 	Config  *types.VirtualMachineConfigInfo
 	Runtime *types.VirtualMachineRuntimeInfo
+
+	newStateEvents map[State]chan struct{}
 }
 
 func NewContainer(id uid.UID) *Handle {
 	con := &Container{
-		ExecConfig: &executor.ExecutorConfig{},
-		State:      StateCreating,
+		ExecConfig:     &executor.ExecutorConfig{},
+		state:          StateCreating,
+		newStateEvents: make(map[State]chan struct{}),
 	}
 	con.ExecConfig.ID = id.String()
-	return newHandle(con)
+	return newHandle(con, con.state)
```

Comment:
This should not be set to the current container state - the handle.state is a request to transition to that power state, not a record of the current state.